go-controller/pkg/allocator/pod/pod_annotation.go

```diff
@@ -91,13 +91,17 @@ func allocatePodAnnotation(
 
 	// no id allocation
 	var idAllocator id.NamedAllocator
-
+	node, err := kube.GetNode(pod.Spec.NodeName)
```

Comment:
`kube.GetNode(pod.Spec.NodeName)` Will perform a direct API call, can't this be passed as a parameter instead to avoid calling the API on every `allocatePodAnnotation`?

Reply:
Let me try